core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -994,7 +993,8 @@ private FileIO newFileIO(SessionContext context, Map<String, String> properties)
     if (null != ioBuilder) {
       return ioBuilder.apply(context, properties);
     } else {
-      String ioImpl = properties.getOrDefault(CatalogProperties.FILE_IO_IMPL, DEFAULT_FILE_IO_IMPL);
+      String ioImpl = properties.getOrDefault(
+              CatalogProperties.FILE_IO_IMPL, CatalogProperties.DEFAULT_REST_FILE_IO_IMPL);
```

Comment:
I'd roll this change back since the property added to `CatalogProperties` should not be public.